packages/automerge-repo/src/helpers/tests/storage-adapter-tests.ts

```diff
@@ -2,336 +2,376 @@ import { describe, expect, it } from "vitest"
 
 import type { StorageAdapterInterface } from "../../storage/StorageAdapterInterface.js"
 
-export function runStorageAdapterTests({ adapter }: Params) {
-  describe("load", () => {
-    it("should return undefined if there is no data", async () => {
-      expect(
-        await adapter.load([
-          "3xuJ5sVKdBaYS6uGgGJH1cGhBLiC",
-          "sync-state",
-          "d99d4820-fb1f-4f3a-a40f-d5997b2012cf",
-        ])
-      ).toStrictEqual(undefined)
+export function runStorageAdapterTests(_setup: SetupFn, title?: string): void {
+  const setup = async () => {
+    const { adapter, teardown = NO_OP } = await _setup()
+    return { adapter, teardown }
+  }
+
+  describe(`Network adapter acceptance tests ${
+    title ? `(${title})` : ""
+  }`, () => {
+    describe("load", () => {
+      it("should return undefined if there is no data", async () => {
+        const { adapter, teardown } = await setup()
+        expect(
+          await adapter.load([
+            "3xuJ5sVKdBaYS6uGgGJH1cGhBLiC",
+            "sync-state",
+            "d99d4820-fb1f-4f3a-a40f-d5997b2012cf",
+          ])
+        ).toStrictEqual(undefined)
+        teardown()
```

Comment:
The reason I intentionally utilized the `afterEach` hook is because it will be called even if a test fails. The `expect` function `throw`s an error. Maybe it is good to use [`onTestFinished`](https://vitest.dev/api/#ontestfinished) like this:
```javascript
const { adapter, teardown } = await setup()
onTestFinished(() => teardown())
```

Reply:
Ok, currently installed version of vitest is `1.2.2` and it doesn't support the onTestFinished.